### PR TITLE
Fix error message

### PR DESCRIPTION
**Pull request by DeveloperMetal**

### CODE REVIEW

File: shipment_management/public/js/get_rates.js [hunk 1]
```diff
@@ -1,90 +1,109 @@
...
+			function(data) {
+				let selected_service = data.service_types;
+				let service_data = service_dict.find((rate) => rate.label == selected_service);
+
+				if (service_data.fee != 0) {
+					frm.add_child("taxes", {
+						"charge_type": "Actual",
+						"account_head": "Freight and Forwarding Charges - JA",
+						"tax_amount": service_data.fee,
+						"description": `Shipping (${service_data.label}")`
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:99]: Codacy Issue found: [Identifier 'service_data' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/provider_fedex.py [hunk 2]
```diff
@@ -8,12 +8,13 @@
...
 import frappe
-from config.app_config import PRIMARY_FEDEX_DOC_NAME, ExportComplianceStatement
 from frappe import _
 from frappe.utils import cint
 from frappe.utils.file_manager import *
 from frappe.utils.password import get_decrypted_password
-from shipment import check_permission
+
+from shipment_management.config.app_config import PRIMARY_FEDEX_DOC_NAME, ExportComplianceStatement
+from shipment_management.shipment import check_permission
```
Comment (vjFaLk) [inline shipment_management/provider_fedex.py:17]: Codacy Issue found: [Unused check_permission imported from shipment_management.shipment](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 3]
```diff
@@ -1,90 +1,109 @@
...
+	frappe.call({
+		freeze: 1,
+		method: 'shipment_management.shipengine.api.get_rates',
+		args: {
+			doc: frm.doc,
+			estimate: false
+		},
+		callback: function(response) {
+			let options = [];
+			let service_dict = [];
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:56]: Codacy Issue found: [Identifier 'service_dict' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/shipment.py [hunk 4]
```diff
@@ -5,14 +5,14 @@
...
 import frappe
-from config.app_config import PRIMARY_FEDEX_DOC_NAME, FedexTestServerConfiguration, StatusMapFedexAndShipmentNote, SupportedProviderList
 from frappe import _
 from frappe.contacts.doctype.address.address import get_company_address
 from frappe.model.document import get_doc
 from frappe.model.mapper import get_mapped_doc
 from frappe.utils import add_months, now
-from utils import get_country_code
 
+from shipment_management.config.app_config import PRIMARY_FEDEX_DOC_NAME, FedexTestServerConfiguration, StatusMapFedexAndShipmentNote, SupportedProviderList
```
Comment (vjFaLk) [inline shipment_management/shipment.py:14]: Codacy Issue found: [Unused StatusMapFedexAndShipmentNote imported from shipment_management.config.app_config](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/custom_script.js [hunk 5]
```diff
@@ -62,12 +63,12 @@ function create_dialog(frm) {
                 },
                 freeze: 1,
                 callback: function (r) {
-                    sales_orders = []
-                    for (var i = 0; i < frm.doc.items.length; i++) {
+                    let sales_orders = [];
```
Comment (vjFaLk) [inline shipment_management/public/js/custom_script.js:66]: Codacy Issue found: [Identifier 'sales_orders' is not in camel case. (camelcase)](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/shipengine/api.py [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+	else:
+		rates = get_shipping_rates(from_address, to_address, package, doc, items, confirmation)
+
+	if isinstance(rates, dict) and rates.get("errors"):
+		frappe.throw(rates.get("errors"))
+
+	# process all the returned rates
+	if not rates:
+		frappe.throw("Could not get rates, please re-check your shipping address.")
+
```
Comment (vjFaLk) [inline shipment_management/shipengine/api.py:123]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 7]
```diff
@@ -1,90 +1,109 @@
...
-                    frm.set_value("fedex_shipping_method", service_dict[data.service_types].label.replace(/ /g, "_"))
-                    refresh_field("taxes");
-                    frm.save()
-                },
-                "Select Service")
-        }
-    })
+			// form all the available shipping options
+			$.each(response.message, function(index, value) {
+				let estimated_arrival = isNaN(value.estimated_arrival) ? `(${value.estimated_arrival})` : `(${value.days} days)`;
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:60]: Codacy Issue found: [Identifier 'estimated_arrival' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 8]
```diff
@@ -1,90 +1,109 @@
...
+				if (service_data.fee != 0) {
+					frm.add_child("taxes", {
+						"charge_type": "Actual",
+						"account_head": "Freight and Forwarding Charges - JA",
+						"tax_amount": service_data.fee,
+						"description": `Shipping (${service_data.label}")`
+					});
+				};
+
+				frm.set_value("fedex_shipping_method", service_data.code);
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:103]: Codacy Issue found: [Identifier 'service_data' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 9]
```diff
@@ -1,90 +1,109 @@
...
+				let service_data = service_dict.find((rate) => rate.label == selected_service);
+
+				if (service_data.fee != 0) {
+					frm.add_child("taxes", {
+						"charge_type": "Actual",
+						"account_head": "Freight and Forwarding Charges - JA",
+						"tax_amount": service_data.fee,
+						"description": `Shipping (${service_data.label}")`
+					});
+				};
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:101]: Codacy Issue found: [Unnecessary semicolon.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 10]
```diff
@@ -1,90 +1,109 @@
...
+				"days": null,
+				"arrival": null
+			});
+
+			// display all the shipping options
+			frappe.prompt({
+				"label": "Service Types",
+				"fieldtype": "Select",
+				"fieldname": "service_types",
+				"options": options,
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:87]: Codacy Issue found: [Expected property shorthand.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 11]
```diff
@@ -1,90 +1,109 @@
...
-                    refresh_field("taxes");
-                    frm.save()
-                },
-                "Select Service")
-        }
-    })
+			// form all the available shipping options
+			$.each(response.message, function(index, value) {
+				let estimated_arrival = isNaN(value.estimated_arrival) ? `(${value.estimated_arrival})` : `(${value.days} days)`;
+				let option = `${value.label} - $${value.fee} ${estimated_arrival}`;
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:61]: Codacy Issue found: [Identifier 'estimated_arrival' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 12]
```diff
@@ -1,90 +1,109 @@
...
+			},
+			function(data) {
+				let selected_service = data.service_types;
+				let service_data = service_dict.find((rate) => rate.label == selected_service);
+
+				if (service_data.fee != 0) {
+					frm.add_child("taxes", {
+						"charge_type": "Actual",
+						"account_head": "Freight and Forwarding Charges - JA",
+						"tax_amount": service_data.fee,
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:98]: Codacy Issue found: [Identifier 'service_data' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 13]
```diff
@@ -1,90 +1,109 @@
...
+			frappe.prompt({
+				"label": "Service Types",
+				"fieldtype": "Select",
+				"fieldname": "service_types",
+				"options": options,
+				"reqd": 1
+			},
+			function(data) {
+				let selected_service = data.service_types;
+				let service_data = service_dict.find((rate) => rate.label == selected_service);
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:92]: Codacy Issue found: [Expected '===' and instead saw '=='.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/custom_script.js [hunk 14]
```diff
@@ -24,20 +24,21 @@ frappe.ui.form.on("Delivery Note", {
...
     }
 
-    var item_count = 1;
-    item_dict = {}
-    for (var i = 0; i < frm.doc.items.length; i++) {
-        for (var j = 0; j < frm.doc.items[i].qty; j++) {
+    let item_count = 1;
+    let item_dict = {}
+    for (let i = 0; i < frm.doc.items.length; i++) {
+        for (let j = 0; j < frm.doc.items[i].qty; j++) {
```
Comment (vjFaLk) [inline shipment_management/public/js/custom_script.js:41]: Codacy Issue found: [Generic Object Injection Sink (security/detect-object-injection)](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/custom_script.js [hunk 15]
```diff
@@ -24,20 +24,21 @@ frappe.ui.form.on("Delivery Note", {
...
             count++;
         }
     }
 
-    var item_count = 1;
-    item_dict = {}
-    for (var i = 0; i < frm.doc.items.length; i++) {
-        for (var j = 0; j < frm.doc.items[i].qty; j++) {
+    let item_count = 1;
+    let item_dict = {}
```
Comment (vjFaLk) [inline shipment_management/public/js/custom_script.js:39]: Codacy Issue found: [Missing semicolon. (semi)](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 16]
```diff
@@ -1,90 +1,109 @@
...
-                option = value.label + " - $" + value.fee + "\n"
-                options_string += option
-                service_dict[option.slice(0, -1)] = {
-                    "label": value.label,
-                    "fee": value.fee
-                }
-            })
+	frm.doc.taxes.forEach((item) => {
+		if (item.account_head == "Freight and Forwarding Charges - JA") {
+			frappe.throw("Shipment Charge has already been added");
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:43]: Codacy Issue found: ['frappe' is not defined.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/custom_script.js (hunk 15 above)
Comment (vjFaLk) [inline shipment_management/public/js/custom_script.js:39]: Codacy Issue found: [Identifier 'item_dict' is not in camel case. (camelcase)](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js (hunk 13 above)
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:92]: Codacy Issue found: [Identifier 'selected_service' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/custom_script.js [hunk 17]
```diff
@@ -62,12 +63,12 @@ function create_dialog(frm) {
...
                 callback: function (r) {
-                    sales_orders = []
-                    for (var i = 0; i < frm.doc.items.length; i++) {
+                    let sales_orders = [];
+                    for (let i = 0; i < frm.doc.items.length; i++) {
                         sales_orders.push(frm.doc.items[i].against_sales_order)
                     }
                     sales_orders = Array.from(new Set(sales_orders));
-                    for (var i = 0; i < sales_orders.length; i++) {
+                    for (let i = 0; i < sales_orders.length; i++) {
```
Comment (vjFaLk) [inline shipment_management/public/js/custom_script.js:71]: Codacy Issue found: [Identifier 'sales_orders' is not in camel case. (camelcase)](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/public/js/get_rates.js [hunk 18]
```diff
@@ -1,90 +1,109 @@
...
+			// display all the shipping options
+			frappe.prompt({
+				"label": "Service Types",
+				"fieldtype": "Select",
+				"fieldname": "service_types",
+				"options": options,
+				"reqd": 1
+			},
+			function(data) {
+				let selected_service = data.service_types;
```
Comment (vjFaLk) [inline shipment_management/public/js/get_rates.js:91]: Codacy Issue found: [Identifier 'selected_service' is not in camel case.](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

File: shipment_management/shipengine/api.py [hunk 19]
```diff
@@ -114,10 +114,13 @@ def get_shipengine_rates(from_address, to_address, items=None, doc=None, estimat
...
 		rates = get_shipping_rates(from_address, to_address, package, doc, items, confirmation)
 
+	if isinstance(rates, dict) and rates.get("errors"):
+		frappe.throw(rates.get("errors"))
+
 	# process all the returned rates
 	if not rates:
 		frappe.throw("Could not get rates, please re-check your shipping address.")
-
+
```
Comment (vjFaLk) [inline shipment_management/shipengine/api.py:123]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/mail_67/shipment_management/pullRequest?prid=5453114)

**Review by Alchez — APPROVED** [review]

**Pull request merged**